src/System.Security.Cryptography.Hashing.Algorithms/tests/HmacTests.cs

```diff
@@ -31,7 +31,14 @@ protected HmacTests(byte[][] testKeys, byte[][] testData)
 
             using (HMAC hmac = Create())
             {
-                hmac.Key = _testKeys[testCaseId];
+                byte[] key = _testKeys[testCaseId];
+                hmac.Key = key;
+                Assert.NotSame(key, hmac.Key);
+                Assert.NotSame(hmac.Key, hmac.Key);
```

Comment:
I'm suggesting guarding against
```
public byte[] Key
{
    get { return _key.CloneByteArray(); }
    set { _key = value; }
}
```

key and hmac.get_Key() will not ReferenceEquals, but hmac is still holding a reference to the original array, so it's subject to mutation errors.

It'd be a very strange implementation, but a lot of tests are about making sure we didn't implement things strangely :).